linker/Linker.Steps/MarkStep.cs

```diff
@@ -131,13 +131,7 @@ void Process ()
 			if (QueueIsEmpty ())
 				throw new InvalidOperationException ("No entry methods");
 
-			ProcessEntireQueue ();
-
-			// After all types have been processed then we can process the lazily marked attributes
-			ProcessLazyAttributes ();
-
-			// We need to process the queue again in case marking the attributes enqueued more work
-			ProcessEntireQueue ();
+			while (ProcessAllQueues ())
```

Comment:
I think `while (ProcessPrimaryQueue () || ProcessLazyAttributes ())` would be better

Reply:
This PR is serving two purposes, (1) fix an obscure bug, (2) set the stage for another enhancement I'm working on.   In this next change I'm working on, I may end up introducing another queue that needs to be processed.  I thought it would be nice to establish a method for handling how we process the queues in this simple PR.  

Would you still prefer the `||` approach if there were 3 methods?

Reply:
I'll see but in this PR `||` approach is much better

Reply:
Done